Attribute/WithLoggerChannel.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MonologBundle\Attribute;
+
+/**
+ * Autoconfigures the logger channel used when inject a logger in the service.
+ *
+ * This adds the `monolog.logger` tag on the service definition.
+ */
+#[\Attribute(\Attribute::TARGET_CLASS)]
+class WithLoggerChannel
+{
+    private string $channel;
```

Comment:
what about using a public property marked as `@readonly`?

Reply:
Done.

I also made the class final as there is no reason to extend it.